client/service_proxy.go

```diff
@@ -22,253 +22,16 @@
-	return wh.client.RespondWorkflowTaskFailed(ctx, req)
-}
-
-func (wh *workflowServiceProxyServer) ScanWorkflowExecutions(ctx context.Context, req *workflowservice.ScanWorkflowExecutionsRequest) (*workflowservice.ScanWorkflowExecutionsResponse, error) {
-	return wh.client.ScanWorkflowExecutions(ctx, req)
-}
-
-func (wh *workflowServiceProxyServer) SignalWithStartWorkflowExecution(ctx context.Context, req *workflowservice.SignalWithStartWorkflowExecutionRequest) (*workflowservice.SignalWithStartWorkflowExecutionResponse, error) {
-	return wh.client.SignalWithStartWorkflowExecution(ctx, req)
-}
-
-func (wh *workflowServiceProxyServer) SignalWorkflowExecution(ctx context.Context, req *workflowservice.SignalWorkflowExecutionRequest) (*workflowservice.SignalWorkflowExecutionResponse, error) {
-	return wh.client.SignalWorkflowExecution(ctx, req)
-}
-
-func (wh *workflowServiceProxyServer) StartBatchOperation(ctx context.Context, req *workflowservice.StartBatchOperationRequest) (*workflowservice.StartBatchOperationResponse, error) {
-	return wh.client.StartBatchOperation(ctx, req)
-}
-
-func (wh *workflowServiceProxyServer) StartWorkflowExecution(ctx context.Context, req *workflowservice.StartWorkflowExecutionRequest) (*workflowservice.StartWorkflowExecutionResponse, error) {
-	return wh.client.StartWorkflowExecution(ctx, req)
-}
-
-func (wh *workflowServiceProxyServer) StopBatchOperation(ctx context.Context, req *workflowservice.StopBatchOperationRequest) (*workflowservice.StopBatchOperationResponse, error) {
-	return wh.client.StopBatchOperation(ctx, req)
-}
-
-func (wh *workflowServiceProxyServer) TerminateWorkflowExecution(ctx context.Context, req *workflowservice.TerminateWorkflowExecutionRequest) (*workflowservice.TerminateWorkflowExecutionResponse, error) {
-	return wh.client.TerminateWorkflowExecution(ctx, req)
-}
-
-func (wh *workflowServiceProxyServer) UpdateNamespace(ctx context.Context, req *workflowservice.UpdateNamespaceRequest) (*workflowservice.UpdateNamespaceResponse, error) {
-	return wh.client.UpdateNamespace(ctx, req)
-}
-
-func (wh *workflowServiceProxyServer) UpdateSchedule(ctx context.Context, req *workflowservice.UpdateScheduleRequest) (*workflowservice.UpdateScheduleResponse, error) {
-	return wh.client.UpdateSchedule(ctx, req)
-}
-
-func (wh *workflowServiceProxyServer) UpdateWorkerBuildIdOrdering(ctx context.Context, req *workflowservice.UpdateWorkerBuildIdOrderingRequest) (*workflowservice.UpdateWorkerBuildIdOrderingResponse, error) {
-	return wh.client.UpdateWorkerBuildIdOrdering(ctx, req)
-}
-
-func (wh *workflowServiceProxyServer) UpdateWorkflow(ctx context.Context, req *workflowservice.UpdateWorkflowRequest) (*workflowservice.UpdateWorkflowResponse, error) {
-	return wh.client.UpdateWorkflow(ctx, req)
+func NewWorkflowServiceProxyServer(options proxy.WorkflowServiceProxyOptions) (workflowservice.WorkflowServiceServer, error) {
```

Comment:
Hrmm, technically a BC break. If we're gonna wrap the `proxy` package's `NewWorkflowSeriviceProxyServer` call, we might as well translate options.

Reply:
Yeah, couldn't decide here. Now put the client type back and casting.

Reply:
> and casting.

Not cast, just transform